pkg/cmd/config/config.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2024 The KitOps Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+package config
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"kitops/pkg/output"
+	"os"
+	"path/filepath"
+	"reflect"
+
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+)
+
+type Config struct {
+	LogLevel string `json:"logLevel"`
+	Progress string `json:"progress"`
+}
+
+// DefaultConfig returns a Config struct with default values.
+func DefaultConfig() *Config {
+	return &Config{
+		LogLevel: output.LogLevelInfo.String(),
+		Progress: "plain",
+	}
+}
+
+// Set a configuration key and value.
+func setConfig(_ context.Context, opts *configOptions) error {
+	configPath := getConfigPath(opts.profile)
+	cfg, err := LoadConfig(configPath)
+	if err != nil {
+		cfg = DefaultConfig()
+	}
+
+	caser := cases.Title(language.Und)
+	v := reflect.ValueOf(cfg).Elem().FieldByName(caser.String(opts.key))
+	if !v.IsValid() {
+		return fmt.Errorf("unknown configuration key: %s", opts.key)
+	}
```

Comment:
Caser is causing issues here; since we're using camelcase for config settings (`LogLevel`), attempting to modify case will mangle fields, turning `LogLevel`, `logLevel`, etc. into `Loglevel`

Reply:
This has been fixed!
![image](https://github.com/user-attachments/assets/8a14e48d-dfd3-40cc-9a59-0ec4db7f2104)